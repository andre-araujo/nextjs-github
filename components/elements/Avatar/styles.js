import styled from 'styled-components';

export const Wrapper = styled.div`
    text-align: center;
`;

export const Image = styled.img`
    width: 100%;
    height: auto;
    border-radius: 5%;
`;
