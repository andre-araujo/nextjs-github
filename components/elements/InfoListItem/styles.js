import styled from 'styled-components';

export const Wrapper = styled.li`
    text-align: center;
    margin-bottom: 1rem;
`;

export const Title = styled.h3`
    font-family: AvantGardeMedium;
    font-size: 1.2rem;
    margin-right: .2rem;
    display: inline-block;
`;

export const Description = styled.p`
    font-family: AvantGardeLight;
    font-size: 1rem;
    display: inline-block;
`;
