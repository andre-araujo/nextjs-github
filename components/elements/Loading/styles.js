import styled, { css, keyframes } from 'styled-components';

const animateDot = keyframes`
    0% {
        opacity: 0;
        transform: rotate(-90deg);
    }

    50% {
        opacity: 1;
        transform: rotate(0);
    }
`;

export default styled.div`
    align-items: center;
    background: rgba(255, 255, 255, 0.85);
    bottom: 0;
    display: flex;
    left: 0;
    opacity: 0;
    pointer-events: none;
    position: fixed;
    justify-content: center;
    right: 0;
    top: 0;
    z-index: 10;

    ${({ active }) => active && css`
        opacity: 1;
        pointer-events: all;
    `}

    span {
        animation: ${animateDot} 1s infinite;
        background: #666;
        display: inline-flex;
        height: 1rem;
        margin: 0.5rem;
        opacity: 0;
        transform: rotate(90deg);
        width: 1rem;
    }

    span:nth-child(2) {
        animation-delay: 0.1s;
    }

    span:nth-child(3) {
        animation-delay: 0.2s;
    }
`;
