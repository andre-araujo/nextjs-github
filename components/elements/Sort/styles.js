import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    font-size: .8rem;
    display: flex;
    user-select: none;
`;

export const Label = styled.span`
    font-family: AvantGardeDemi;
    margin-right: .6rem;
`;

export const Option = styled.li`
    font-family: AvantGardeMedium;
    display: inline-block;

    ${({ active }) => active && css`
        color: #F1912D;
    `}

    :hover {
        ${({ active }) => !active && css`
            color: #ffcc98;
            cursor: pointer;
        `}
    }

    :not(:last-child) {
        padding-right: .6rem;
        margin-right: .6rem;

        border-right: 1px solid #333;
    }
`;
