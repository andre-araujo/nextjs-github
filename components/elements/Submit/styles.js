import styled from 'styled-components';

export default styled.input`
    background: transparent;
    border: 1px solid #333;
    color: #333;
    cursor: pointer;
    display: inline-block;
    font-family: AvantGardeDemi;
    font-size: 1.25rem;
    margin: 0;
    outline: none;
    padding: 0.625rem 2rem;
    transition: 0.2s background, 0.2s color;

    :hover {
        background: #333;
        color: #fafafa;
    }
`;
