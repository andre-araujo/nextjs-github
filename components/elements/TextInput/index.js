
import PropTypes from 'prop-types';

import {
    Wrapper,
    Content,
    Label,
    Field,
} from './styles';

function TextInput({
    label,
    ...inputProps
}) {
    return (
        <Wrapper>
            <Content>
                <Field
                    {...inputProps}
                    type="text"
                    required
                    name={inputProps.name}
                />
                <Label htmlFor={inputProps.name}>
                    {label}
                </Label>
            </Content>
        </Wrapper>
    );
}

TextInput.defaultProps = {
    name: null,
    onChange: null,
    label: 'label',
};

TextInput.propTypes = {
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
};

export default TextInput;
