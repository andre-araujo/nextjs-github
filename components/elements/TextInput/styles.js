import styled from 'styled-components';

export const Wrapper = styled.div`
    padding: 1.25rem 0;
`;

export const Content = styled.div`
    position: relative;
    border-bottom: 1px solid #333;
`;

export const Label = styled.label`
    color: #333;
    font-family: AvantGardeLight;
    font-size: 1.5rem;
    left: 0;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    transition: 0.2s top, 0.2s transform, 0.2s font-size;
    pointer-events: none;
`;

export const Field = styled.input`
    background: none;
    border: none;
    box-shadow: none;
    color: #333;
    font-family: AvantGardeLight;
    font-size: 1.5rem;
    outline: none;
    width: 100%;

    :focus ~ ${Label},
    :valid ~ ${Label} {
        font-size: 1rem;
        top: 0;
        transform: translateY(-100%);
    }
`;
