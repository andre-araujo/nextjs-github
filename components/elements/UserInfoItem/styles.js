import styled from 'styled-components';

export const Wrapper = styled.li`
    margin: 1rem 0;
`;

export const Title = styled.h3`
    font-family: AvantGardeDemi;
    font-size: 1rem;
    margin-bottom: .5em;
`;


export const Description = styled.p`
    font-size: .8rem;
    margin-right: .2rem;
    font-family: AvantGardeLight;
`;
