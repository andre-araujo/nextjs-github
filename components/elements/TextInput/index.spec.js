import React from 'react';
import {
    mount,
} from 'enzyme';

import TextInput from './index';
import { Field } from './styles';

describe('TextInput component', () => {
    let props;

    beforeEach(() => {
        props = {
            name: 'nameMock',
            label: 'labelMock',
            onChange: jest.fn(),
        };
    });

    it('should match snapshot', () => {
        expect(mount(
            <TextInput {...props} />,
        )).toMatchSnapshot();
    });

    it('should call onChange function', () => {
        const Input = mount(
            <TextInput {...props} />,
        );

        Input
            .find(Field)
            .simulate('change', { target: { value: 'test' } });

        expect(props.onChange).toHaveBeenCalled();
    });
});
