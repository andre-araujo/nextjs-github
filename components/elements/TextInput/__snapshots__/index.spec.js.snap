// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component should match snapshot 1`] = `
<div
  className="sc-bdVaJa ibpmVy"
>
  <div
    className="sc-bwzfXH hTErTh"
  >
    <input
      className="sc-bxivhb kIdduM"
      name="nameMock"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      className="sc-htpNat iUdHjJ"
      htmlFor="nameMock"
    >
      labelMock
    </label>
  </div>
</div>
`;
